modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/model/listener/CompanyModelListener.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.js.importmaps.extender.internal.model.listener;
+
+import com.liferay.frontend.js.importmaps.extender.internal.servlet.taglib.JSImportMapsExtenderTopHeadDynamicInclude;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(service = ModelListener.class)
+public class CompanyModelListener extends BaseModelListener<Company> {
+
+	@Override
+	public void onAfterCreate(Company company) {
+		JSImportMapsExtenderTopHeadDynamicInclude
+			jsImportMapsExtenderTopHeadDynamicInclude =
+				(JSImportMapsExtenderTopHeadDynamicInclude)_dynamicInclude;
+
+		jsImportMapsExtenderTopHeadDynamicInclude.invalidateCache();
```

Comment:
@izaera I do not understand why you are doing this.

When I am still creating my current company, the cache is for sure does not have any entries related to the current company, and the cache values are company aware now, current cache values should not be affected by this new company either.

Then why need to invalidate?

Reply:
>the cache is for sure does not have any entries related to the current company,

Yes, that's why I invalidate it, so that it is recreated when it is requested. Otherwise this will return `null` and eveything else will fail -> https://github.com/liferay-core-infra/liferay-portal/pull/7705/files#diff-01615da07aface02e8a024c85411ed31fd569d753a1896a4b83bf1b58be85610R37

Given how fast is the recreation of the cache information (it's just a few string concatenations, so not a big deal) we can invalidate the whole cache favoring simpler code at the cost of losing an insignificant optimization.

The `cache validity` is a function of both registered `companies` and registered `import maps`. Thus, on any change of those two entities, the cache is no longer formally valid.

I'm reopening the PR again. Feel free to ellaborate on this if you think my reasoning is flawed.

Reply:
I'm thinking now that since I'm not using a public API to invalidate the cache, I can add a method to compute just the changed company. In a previous PR I had to add a public method for that so I preferred to keep the contract clean than optimizing, but in this case we can have both, I think :+1: .